src/components/DoNotSellAcknowledgement.tsx

```diff
@@ -0,0 +1,79 @@
+// external
+import { h, JSX } from 'preact';
+import { useIntl } from 'react-intl';
+
+// global
+import { noticeAndDoNotSellMessages } from '../messages';
+import type { HandleSetViewState, NavigatorWithGPC } from '../types';
+import { useAirgap } from '../hooks';
+
+// local
+import Button from './Button';
+import ColumnContent from './ColumnContent';
+import Title from './Title';
+import { Purpose } from '@transcend-io/airgap.js-types';
+import { useEffect } from 'preact/hooks';
+
+/**
+ * Component showing acknowledgement of do not sell
+ */
+export default function DoNotSellAcknowledgement({
+  handleSetViewState,
+}: {
+  /** Function to change viewState */
+  handleSetViewState: HandleSetViewState;
+}): JSX.Element {
+  const { airgap } = useAirgap();
+  const { formatMessage } = useIntl();
+
+  const handleOptOut: JSX.MouseEventHandler<HTMLButtonElement> | undefined = (
+    event: JSX.TargetedEvent<HTMLButtonElement, MouseEvent>,
+  ): void => {
+    event.preventDefault();
```

Comment:
@eligrey the new design for this modal requires us to set the consent preferences on footer click. im not sure we have access to an event here, how can we update the consent here without an event? or do we need to pass the event through the transcend.showConsentManager function?

Reply:
If by 'footer' you mean 'within the modal', then we definitely should be able to capture the click here.

Reply:
its not within the modal. its the footer on say, the athetic website. and it calls transcend.showConsentManager() on click